src/NuGet.Clients/NuGet.Console/OutputConsole/BuildOutputConsole.cs

```diff
@@ -36,25 +36,32 @@ public BuildOutputConsole(IVsOutputWindow vsOutputWindow)
 
         public int ConsoleWidth => DefaultConsoleWidth;
 
-        [SuppressMessage("Microsoft.VisualStudio.Threading.Analyzers", "VSTHRD010", Justification = "NuGet/Home#4833 Baseline")]
         public void Activate()
         {
-            if (_outputWindowPane == null)
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
             {
-                _vsOutputWindow.GetPane(ref BuildWindowPaneGuid, out _outputWindowPane);
-            }
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                if (_outputWindowPane == null)
```

Comment:
Does `Activate` actually get called from a background thread such that this is necessary? I would expect given it's a window activation event (at least, it looks like it) that it would always be invoked from the UI thread, in which case I would tend to use `ThreadHelper.ThrowIfNotOnUIThread()` and avoid the delegate.
Either way is fine though.

Reply:
I'd imagine it'll always be called on UI thread but wasn't sure of it. So where ever I wasn't sure I took the more defensive approach instead of relying on underneath system.